### PR TITLE
cpdp-1014: added metrics for grafana

### PR DESCRIPTION
**Pull request by AndyHWChung**

### CODE REVIEW

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/metrics/Metrics.scala [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  val ivTechnicalIssueCounter: Counter = counter("iv.journey-status.technical-issue.count")
+
+  val ivTimeoutCounter: Counter = counter("iv.journey-status.timeout.count")
+
+  val ivUnknownErrorCounter: Counter = counter("iv.journey-status.unknown-error.count")
+  
+  val ivGetFailedJourneyStatusTimer: Timer = timer("iv.get-failed-journey-status.time")
+
+  val ivGetFailedJourneyStatusErrorCounter: Counter = counter("iv.get-failed-journey-status.errors.count")
+
```
Comment (co-category) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/metrics/Metrics.scala]: might be worth countring how many users come through with a wrong-gg-account and also organisations which want to report a tax for a trust.

Comment (AndyHWChung) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/metrics/Metrics.scala]: good idea, I'll add those in

Comment (co-category) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/metrics/Metrics.scala]: also the number of users we push to register with the Trust service so we get an idea of how many times that happens

Comment (AndyHWChung) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/metrics/Metrics.scala]: I've added in a few more metrics now for what I think it was you were describing

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/onboarding/DeterminingIfOrganisationIsTrustController.scala [hunk 2]
```diff
@@ -88,29 +90,32 @@ class DeterminingIfOrganisationIsTrustController @Inject()(
...
-                updateSession(sessionStore, request)(
-                  _.copy(
-                    journeyStatus = Some(
-                      DeterminingIfOrganisationIsTrust(
-                        determiningIfOrganisationIsTrust.ggCredId,
-                        Some(isReportingForTrust),
-                        None
-                      )
+              BadRequest(
+                doYouWantToReportForATrustPage(formWithError, controllers.routes.StartController.weNeedMoreDetails())
```
Comment (co-category) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/onboarding/DeterminingIfOrganisationIsTrustController.scala]: looks like you hit return here by accident

Comment (AndyHWChung) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/onboarding/DeterminingIfOrganisationIsTrustController.scala]: I think that was scalafmt making the long line shorter

**Pull request merged**